codalab/worker/image_manager.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import threading
+import time
+import traceback
+from collections import namedtuple
+
+from codalab.worker.worker_thread import ThreadDict
+
+logger = logging.getLogger(__name__)
+# Stores the download state of a Docker image (also includes the digest being pulled, digest string,
+# DependencyStage and relevant status message from the download)
+ImageAvailabilityState = namedtuple('ImageAvailabilityState', ['digest', 'stage', 'message'])
+
+
+class ImageManager:
+    """
+    ImageManager is the base interface for all image managers (docker, singularity, etc.)
+    No actual logic is encapsulated in this class.
+    """
+
+    def __init__(self, max_image_size: int, max_image_cache_size: int):
```

Comment:
Add docstrings for all methods -- see https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html for format